includes/abstracts/abstract-wc-product.php

```diff
@@ -1102,6 +1153,56 @@ public function set_downloads( $raw_downloads ) {
 		}
 	}
 
+	/**
+	 * Set gallery attachment ids.
+	 *
+	 * @since 2.7.0
+	 * @param array $gallery_ids
+	 */
+	public function set_gallery_attachment_ids( $gallery_ids ) {
+		$this->data['gallery_attachment_ids'] = $gallery_ids;
+	}
+
+	/**
+	 * Set download limit.
+	 *
+	 * @since 2.7.0
+	 * @param int $download_limit
+	 */
+	public function set_download_limit( $download_limit ) {
+		$this->data['download_limit'] = -1 === (int) $download_limit || '' === $download_limit ? -1 : absint( $download_limit );
+	}
+
+	/**
+	 * Set download expiry.
+	 *
+	 * @since 2.7.0
+	 * @param int $download_expiry
+	 */
+	public function set_download_expiry( $download_expiry ) {
+		$this->data['download_expiry'] = -1 === (int) $download_expiry || '' === $download_expiry ? -1 : absint( $download_expiry );
+	}
+
+	/**
+	 * Set download type.
+	 *
+	 * @since 2.7.0
+	 * @param string $download_type
+	 */
+	public function set_download_type( $download_type ) {
```

Comment:
Looks like the new schema handler doesn't use or need this prop - in a comment the author said Google expects product type regardless of what is being sold. We may be able to drop.